tests/frontend/io/test_io_xir.py

```diff
@@ -0,0 +1,646 @@
+        ops.BSgate(0.45, np.pi/2) | (q[0], q[2])
+        ops.Sgate(3*np.pi/2, 0) | q[1]
+        ops.BSgate(2*np.pi, 0.62) | (q[0], q[1])
+        ops.MeasureFock() | q[0]
+"""
+)
+
+prog_txt_tdm = inspect.cleandoc(
+    """
+    import strawberryfields as sf
+    from strawberryfields import ops
+
+    prog = sf.TDMProgram(N=[2, 3])
+    eng = sf.Engine("gaussian")
+
+    with prog.context([np.pi, 3*np.pi/2, 0], [1, 0.5, np.pi], [0, 0, 0]) as (p, q):
+        ops.Sgate(0.123, np.pi/4) | q[2]
+        ops.BSgate(p[0], 0.0) | (q[1], q[2])
+        ops.Rgate(p[1]) | q[2]
+        ops.MeasureHomodyne(p[0]) | q[0]
+        ops.MeasureHomodyne(p[2]) | q[2]
+
+    results = eng.run(prog)
+"""
+)
+
+
+class TestSaveXIR:
+    """Test sf.save functionality"""
+
+    def test_save_filename_path_object(self, prog, tmpdir):
+        """Test saving a program to a file path using a path object"""
+        filename = tmpdir.join("test.xbb")
+        sf.save(filename, prog, ir="xir")
+
+        with open(filename, "r") as f:
+            res = f.read()
+
+        assert res == test_xir_prog_not_compiled
+
+    def test_save_filename_string(self, prog, tmpdir):
+        """Test saving a program to a file path using a string filename"""
+        filename = str(tmpdir.join("test.xbb"))
+        sf.save(filename, prog, ir="xir")
+
+        with open(filename, "r") as f:
+            res = f.read()
+
+        assert res == test_xir_prog_not_compiled
+
+    def test_save_file_object(self, prog, tmpdir):
+        """Test writing a program to a file object"""
+        filename = tmpdir.join("test.xbb")
+
+        with open(filename, "w") as f:
+            sf.save(f, prog, ir="xir")
+
+        with open(filename, "r") as f:
+            res = f.read()
+
+        assert res == test_xir_prog_not_compiled
+
+    def test_save_append_extension(self, prog, tmpdir):
+        """Test appending the extension if not present"""
+        filename = str(tmpdir.join("test.txt"))
+        sf.save(filename, prog, ir="xir")
+
+        with open(filename + ".xbb", "r") as f:
+            res = f.read()
+
+        assert res == test_xir_prog_not_compiled
+
+
+class TestLoadXIR:
+    """Test sf.load functionality"""
+
+    def test_invalid_file(self, prog, tmpdir):
+        """Test exception is raised if file is invalid"""
+        with pytest.raises(ValueError, match="must be a string, path"):
+            sf.load(1, ir="xir")
+
+    def assert_programs_equal(self, prog1, prog2):
+        """Asserts that two SF Program objects are equivalent"""
+        for cmd1, cmd2 in zip(prog1.circuit, prog2.circuit):
+            assert cmd1.op.__class__.__name__ == cmd2.op.__class__.__name__
+            assert cmd1.reg[0].ind == cmd2.reg[0].ind
+            if cmd1.op.p:
+                assert np.all(cmd1.op.p[0] == cmd2.op.p[0])
+
+    def test_load_filename_path_object(self, prog, tmpdir):
+        """Test loading a program using a path object"""
+        filename = tmpdir.join("test.xbb")
```

Comment:
```suggestion
        filename = tmpdir.join("test.xir")
```
Same comment as above regarding test coverage.